src/dataServices/dynamoDbDataService.ts

```diff
@@ -107,6 +107,9 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
     async vReadResource(request: vReadResourceRequest): Promise<GenericResponse> {
         this.assertValidTenancyMode(request.tenantId);
         const { resourceType, id, vid, tenantId } = request;
+        if (Number.isNaN(Number(vid))) {
+            throw new UnauthorizedError('Invalid versionId');
```

Comment:
Is it unauthorized or bad request?

Reply:
From review we want unauthorized for this sort of request

Reply:
But description says it should be 400 - Bad request (which makes sense); Unauthorized is 403. If `throw new UnauthorizedError` is mapped to 400, this is something we should address

Reply:
Apparently it is 403: https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/router/routes/errorHandling.ts#L35 .
Either we should add new error class to be mapped to 400 or rename `isInvalidSearchParameterError` to be more generic - this might be a bigger footprint on number of changed files, so might be former is better